### PR TITLE
docs: Add docs example of how to use tags with feature views

### PR DESCRIPTION
**Pull request by tmihalac**
# What this PR does / why we need it:
Added documentation with examples of how to use the tags in the cli, in the definition file and in the code with feature views.

# Which issue(s) this PR fixes:
Fixes https://github.com/feast-dev/feast/issues/4534

### CODE REVIEW

File: docs/getting-started/concepts/tags.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+The same example of listing feature-views without tag filtering:
+```commandline
+$ feast feature-views list
+NAME                       ENTITIES    TYPE
+driver_hourly_stats        {'driver'}  FeatureView
+driver_hourly_stats_fresh  {'driver'}  FeatureView
+transformed_conv_rate_fresh  {'driver'}  OnDemandFeatureView
+transformed_conv_rate        {'driver'}  OnDemandFeatureView
+```
+
```
Comment (redhatHameed) [inline docs/getting-started/concepts/tags.md:59]: WDYT adding example for permission with tags ? if this relevant here ? 

```
Permission(
    name="team1-fvs",
    types=[FeatureView],
    required_tags={"team": "team1"},
    actions=[CRUD, WRITE, QUERY],
    policy=RoleBasedPolicy(roles=[
"role1"]),
)
```

Comment (tmihalac) [reply docs/getting-started/concepts/tags.md:59]: The jira talks about Feature Views
@dandawg Do you think I should add a permission example ?

File: docs/getting-started/concepts/tags.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Tags
+
+## Overview
+
+Tags is a way to filter Feast objects when listing the objects in the UI, Cli or when directly querying the registry.
```
Comment (lokeshrangineni) [inline docs/getting-started/concepts/tags.md:5]:
```suggestion
Tags in Feast allow for efficient filtering of Feast objects when listing them in the UI, CLI, or querying the registry directly. 
```

Comment (lokeshrangineni) [reply docs/getting-started/concepts/tags.md:5]: @franciscojavierarceo - can we mention in our documentation saying "it provides similar functionality as  `kubernetes labels`" ?

File: docs/getting-started/concepts/tags.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        name="test kafka stream feature view",
+        entities=[entity],
+        schema=[],
+        description="desc",
+        timestamp_field="event_timestamp",
+        source=stream_source,
+        tags={"team": "driver_performance"},
+```
+
+An example of listing feature-views with the tag `team:driver_performance`:
```
Comment (lokeshrangineni) [inline docs/getting-started/concepts/tags.md:42]:
```suggestion
An example of filtering feature-views with the tag `team:driver_performance`:
```

File: docs/getting-started/concepts/tags.md (hunk 1 above)
Comment (redhatHameed) [reply docs/getting-started/concepts/tags.md:59]: in that case, we can leave it.

**Comment by lokeshrangineni** [comment]
/LGTM.
The comments i made are personal opinions. Please feel free to disregard in case if you think them not necessary.

File: docs/getting-started/concepts/tags.md (hunk 2 above)
Comment (tokoko) [reply docs/getting-started/concepts/tags.md:5]: I guess we can, but I don't really see a reason to do that. imho, everyone know what a tag means anyway.

**Review by redhatHameed — APPROVED** [review]

File: docs/getting-started/concepts/tags.md (hunk 2 above)
Comment (franciscojavierarceo) [reply docs/getting-started/concepts/tags.md:5]: Yeah, sure! Feel free to add.

File: docs/getting-started/concepts/tags.md (hunk 3 above)
Comment (franciscojavierarceo) [reply docs/getting-started/concepts/tags.md:42]: I agree here.

**Review by tokoko — APPROVED** [review]

**Pull request merged**